src/libcollections/string.rs

```diff
@@ -1881,6 +1881,41 @@ impl Into<Vec<u8>> for String {
     }
 }
 
+#[stable(feature = "stringfromchars", since = "1.12.0")]
+impl<'a> From<&'a [char]> for String {
+    #[inline]
+    fn from(v: &'a [char]) -> String {
+        let mut s = String::with_capacity(v.len());
```

Comment:
I’m really uninterested in tackling *this* implementation in this PR.

But since you’ve asked, I ought to state my opinion on the topic, at least. I agree that nobody would store ASCII-only text as UTF-32 (there’s bytestrings, after all) and any `ratio > 1` is therefore better than `ratio = 1`. `ratio = 1.5` to `1.6` coupled with the fact that reallocation doubles capacity could be a good choice, especially given the fact that `ratio(jp)`, `ratio(cn)` and `ratio(ko)` are all somewhere between 2 and a bit over 3.

That being said, my gut tells me that nobody would be using this conversion with any serious expectations towards its performance, thus thinking about this problem is not very productive.